src/main/kotlin/no/nav/familie/ef/sak/task/BehandlingsstatistikkTask.kt

```diff
@@ -0,0 +1,151 @@
+        val behandling = behandlingService.hentBehandling(behandlingId)
+        val fagsak = fagsakService.hentFagsak(behandling.fagsakId)
+
+        val sisteOppgaveForBehandling = finnSisteOppgaveForBehandlingen(behandlingId, oppgaveId)
+        val resultatBegrunnelse = finnResultatBegrunnelse(hendelse, behandlingId)
+        val søker = personService.hentSøker(personIdent);
+        val søknadstidspunkt = finnSøknadstidspunkt(fagsak, behandlingId)
+
+        val behandlingsstatistikkDto = BehandlingsstatistikkDto(
+                behandlingId = behandlingId,
+                personIdent = personIdent,
+                gjeldendeSaksbehandlerId = gjeldendeSaksbehandler,
+                eksternFagsakId = fagsak.eksternId.id.toString(),
+                hendelseTidspunkt = hendelseTidspunkt.atZone(zoneIdOslo),
+                søknadstidspunkt = søknadstidspunkt.atZone(zoneIdOslo),
+                hendelse = hendelse,
+                behandlingResultat = behandling.resultat.name,
+                resultatBegrunnelse = resultatBegrunnelse,
+                opprettetEnhet = sisteOppgaveForBehandling.opprettetAvEnhetsnr ?: "9999",
+                ansvarligEnhet = sisteOppgaveForBehandling.tildeltEnhetsnr ?: "9999",
+                strengtFortroligAdresse = søker.adressebeskyttelse.gjeldende()?.gradering == AdressebeskyttelseGradering.STRENGT_FORTROLIG,
+                stønadstype = StønadType.valueOf(fagsak.stønadstype.name),
+                behandlingstype = BehandlingType.valueOf(behandling.type.name)
+        )
+
+        iverksettClient.sendBehandlingsstatistikk(behandlingsstatistikkDto);
+    }
+
+    private fun finnSisteOppgaveForBehandlingen(behandlingId: UUID, oppgaveId: Long?): Oppgave {
+        val gsakOppgaveId = oppgaveId ?: oppgaveService.finnSisteOppgaveForBehandling(behandlingId).gsakOppgaveId
+
+        return oppgaveService.hentOppgave(gsakOppgaveId)
+    }
+
+    private fun finnResultatBegrunnelse(hendelse: Hendelse, behandlingId: UUID): String? {
+        return when (hendelse) {
+            Hendelse.PÅBEGYNT, Hendelse.MOTTATT -> null
+            else -> {
+                val vedtak = vedtakService.hentVedtak(behandlingId)
+                return when (vedtak.resultatType) {
+                    ResultatType.INNVILGE -> vedtak.periodeBegrunnelse
+                    ResultatType.AVSLÅ -> vedtak.avslåBegrunnelse
+                    ResultatType.HENLEGGE -> error("Ikke implementert")
+                }
+            }
+        }
+    }
+
+    private fun finnSøknadstidspunkt(fagsak: Fagsak,
+                                     behandlingId: UUID): LocalDateTime {
+        return when (fagsak.stønadstype) {
+            Stønadstype.OVERGANGSSTØNAD -> søknadService.hentOvergangsstønad(behandlingId).datoMottatt
+            Stønadstype.BARNETILSYN -> søknadService.hentBarnetilsyn(behandlingId).datoMottatt
+            Stønadstype.SKOLEPENGER -> søknadService.hentSkolepenger(behandlingId).datoMottatt
```

Comment:
Hmm, kan vi gjøre dette på en smidigere måte? Å hente hele søknaden er tungt, og er noe sånt 10+ kall mot databasen. Vet vi ikke har noen direkte krav eller behov på ytelse, men anbefaler likevel å legge til en Repository for eks ISøknadsskjema som henter mottattDato eller noe slik

Reply:
Godt innspill! 
La til en egen spørring for dette